File: 1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> tmp;
        for (int i = 0; i < nums.size(); ++i) {
            if (tmp.find(target - nums[i]) != tmp.end()) {
                return {tmp[target-nums[i]], i};
            }
            tmp[nums[i]] = i;
        }
        return {};
    }
};
// @lc code=end

File: 1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python
#
# [1] 两数之和
#

# @lc code=start
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        m = {}
        for (i, v) in enumerate(nums):
            if (target - v) in m.keys():
                return [i, m[target - v]]
            m[v] = i

# @lc code=end
